spec/components/composer_messages_finder_spec.rb

```diff
@@ -467,4 +467,20 @@
     end
   end
 
+  context 'when editing a post' do
+    let(:user) { Fabricate.build(:user) }
+    let(:new_post_finder) { ComposerMessagesFinder.new(user, composer_action: 'reply') }
+    let(:edit_post_finder) { ComposerMessagesFinder.new(user, composer_action: 'edit') }
+
+    before do
+      SiteSetting.educate_until_posts = 10
+    end
+
+    it "returns nothing even if it normally would" do
+      user.expects(:post_count).returns(9)
+      expect(edit_post_finder.find).to be_blank
+      expect(new_post_finder.find).to be_present
```

Comment:
`be_present` is abit vague for me because the method can return anything and the test will still pass. It'll be better if we can test for the exact return value here.

Reply:
I can change this to:

```ruby
expect(new_post_finder.find).to be_instance_of Hash
```

Does that work? Sorry for all the questions here.

```ruby
  def find
    return if editing_post?
    self.class.check_methods.each do |m|
      msg = send(m)
      return msg if msg.present?
    end
    nil
  end
```

Find is just nebulous to me because it calls other methods and they all return different hashes of different forms -- so I'm not entirely sure how to accommodate this request other than making sure it returns a hash.